extensions/standard-processors/tests/integration/InvokeHTTPTests.cpp

```diff
@@ -271,9 +290,12 @@ TEST_CASE("InvokeHTTP Attributes to Send uses full string matching, not substrin
   auto file_contents = result.at(InvokeHTTP::Success);
   REQUIRE(file_contents.size() == 1);
   REQUIRE(test_controller.plan->getContent(file_contents[0]) == "data");
-  REQUIRE(utils::verifyLogLinePresenceInPollTime(1s, "key:header value:value2"));
-  REQUIRE_FALSE(LogTestController::getInstance().contains("key:header1 value:value1"));
-  REQUIRE_FALSE(LogTestController::getInstance().contains("key:invalid", 0s));
+  REQUIRE(http_server.getHeaders().at("header") == "value2");
+  REQUIRE_FALSE(http_server.getHeaders().contains("header1"));
+  auto header_keys = http_server.getHeaderKeys();
+  REQUIRE(std::find_if(header_keys.begin(), header_keys.end(), [] (const auto& key) {
+    return minifi::utils::string::startsWith(key, "invalid");
+  }) == header_keys.end());
```

Comment:
Higher level algorithms make the code more descriptive
```suggestion
  REQUIRE(std::none_of(header_keys.begin(), header_keys.end(), [] (const auto& key) {
    return minifi::utils::string::startsWith(key, "invalid");
  }));
```

This snippet is repeated 4 times, and it's not trivial, so I think it should be extracted.

Reply:
Good point, updated in https://github.com/apache/nifi-minifi-cpp/pull/1929/commits/6c1f137170ca6a00a79b19e3427f9144e1bcefa7